spec/symengine_spec.rb

```diff
@@ -18,5 +18,52 @@
     it { is_expected.to be_a SymEngine::Rational }
     its(:to_s) { is_expected.to eq '1/3' }
   end
+  
+  describe 'parse' do
+    subject { SymEngine::parse('123 + 321') }
+    
+    it { is_expected.to be_a SymEngine::Integer }
+    it { is_expected.to eq 444 }
+  end
+
+  it 'gives parse errors' do
+    expect { SymEngine::parse('12a + n34a9') }.to raise_error(RuntimeError)
```

Comment:
You don't have to create new exceptions.
- `SYMENGINE_RUNTIME_ERROR` - `rb_eRuntimeError`
- `SYMENGINE_DIV_BY_ZERO` - `rb_eZeroDivError`
- `SYMENGINE_NOT_IMPLEMENTED` - `rb_eNotImpError`
- `SYMENGINE_PARSE_ERROR` - `rb_eSyntaxError` 
- `SYMENGINE_UNDEFINED` - `rb_eFloatDomainError` ?? (I'm not sure what `SYMENGINE_UNDEFINED` is. Is this supposed to be a Domain error?)